src/adaptix/_internal/model_tools/introspection/sqlalchemy_tables.py

```diff
@@ -0,0 +1,149 @@
+        return DefaultFactory(factory=column_default.arg.__wrapped__)
+    if isinstance(column_default, ScalarElementColumnDefault):
+        return DefaultValue(value=column_default.arg)
+    return NoDefault()
+
+
+def _get_sqlalchemy_required(column):
+    return not (
+        column.default or column.nullable or column.server_default
+        or (column.primary_key and column.autoincrement and column.type.python_type is int)
+    )
+
+
+def _get_sqlalchemy_input_shape(tp, columns, relationships, type_hints) -> InputShape:
+    param_name_to_field = {
+        column.name: InputField(
+            id=column.name,
+            type=_get_sqlalchemy_type_for_column(column, type_hints),
+            default=_get_sqlalchemy_default(column.default),
+            is_required=_get_sqlalchemy_required(column),
+            metadata=column.info,
+            original=ColumnPropertyWrapper(column_property=column)
+        )
+        for column in columns
+    }
+
+    for relationship in relationships:
+        name = str(relationship).split(".")[1]
+        param_name_to_field[name] = InputField(
+            id=name,
+            type=_get_sqlalchemy_type_for_relationship(relationship, type_hints),
+            default=NoDefault(),
+            is_required=False,
+            metadata={},
+            original=relationship
+        )
+
+    fields = tuple(param_name_to_field.values())
+
+    return InputShape(
+        constructor=tp,
+        fields=fields,
+        overriden_types=frozenset(),
+        kwargs=None,
+        params=tuple(
+            Param(
+                field_id=column.name,
+                name=column.name,
+                kind=ParamKind.KW_ONLY
+            )
+            for column in columns
+        )
+    )
+
+
+def _get_sqlalchemy_output_shape(columns, relationships, type_hints) -> OutputShape:
+    output_fields = [
+        OutputField(
+            id=column.name,
+            type=_get_sqlalchemy_type_for_column(column, type_hints),
+            default=_get_sqlalchemy_default(column.default),
+            metadata=column.info,
+            original=ColumnPropertyWrapper(column_property=column),
+            accessor=create_attr_accessor(column.name, is_required=True)
+        )
+        for column in columns
+    ]
+
+    for relationship in relationships:
+        name = str(relationship).split(".")[1]
+        output_fields.append(
+            OutputField(
+                id=name,
+                type=_get_sqlalchemy_type_for_relationship(relationship, type_hints),
+                default=NoDefault(),
+                metadata={},
+                original=relationship,
+                accessor=create_attr_accessor(name, is_required=False)
+            )
+        )
+
+    return OutputShape(
+        fields=tuple(output_fields),
+        overriden_types=frozenset()
+    )
+
+
+def get_sqlalchemy_shape(tp) -> FullShape:
+    columns = inspect(tp).columns
```

Comment:
If should raise ``NoTargetPackage``, ``PackageIsTooOld`` and ``IntrospectionImpossible``